runtime/src/lib.rs

```diff
@@ -319,24 +328,35 @@ pub(crate) async fn stop(
     Ok(())
 }
 
-fn offer() -> anyhow::Result<Option<serde_json::Value>> {
+fn offer(self_test_result: serde_json::Value) -> anyhow::Result<serde_json::Value> {
     let cpu = CpuInfo::try_new()?;
     let model = format!(
         "Stepping {} Family {} Model {}",
         cpu.model.stepping, cpu.model.family, cpu.model.model
     );
 
-    // TODO add capabilities if GPU supported
-    Ok(Some(serde_json::json!({
+    let mut runtime_capabilities = vec!["inet", "vpn", "manifest-support", "start-entrypoint"];
+    if is_gpu_supported(&self_test_result) {
+        runtime_capabilities.push("!exp:gpu");
```

Comment:
If GPU supported offer will contain experimental `gpu` capability (for now disabled)